### ISSUE TITLE
The purpose of current_page in applications_page_check

### ISSUE THREAD

**Issue by mmarzantowicz** [django-cms/django-cms#3378]
What is the purpose of `current_page` argument in following function (https://github.com/divio/django-cms/blob/develop/cms/appresolver.py#L29)?

``` Python
def applications_page_check(request, current_page=None, path=None):
    """Tries to find if given path was resolved over application.
    Applications have higher priority than other cms pages.
    """
    if current_page:
        return current_page
```

Quick grep over source code shows that it's not used anywhere. What is more, if it is `not None` it returns itself and makes `applications_page_check` quite ... useless?

``` Bash
$ grep -nIR 'applications_page_check'
cms/middleware/page.py:6:    from cms.appresolver import applications_page_check
cms/middleware/page.py:14:            request._current_page_cache = applications_page_check(request)
cms/appresolver.py:28:def applications_page_check(request, current_page=None, path=None):
cms/tests/apphooks.py:10:from cms.appresolver import applications_page_check, clear_app_resolvers, get_app_patterns
cms/tests/apphooks.py:175:            attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash
cms/tests/apphooks.py:187:            attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash and language prefix
cms/tests/apphooks.py:234:                    attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash
cms/tests/apphooks.py:241:                    attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash
cms/tests/apphooks.py:260:            attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash
cms/tests/apphooks.py:277:            attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash
cms/tests/apphooks.py:289:            attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash
cms/tests/apphooks.py:355:            attached_to_page = applications_page_check(request, path=path[1:])  # strip leading slash
cms/views.py:153:        # since we always have a page at this point, applications_page_check is
cms/views.py:155:        # page = applications_page_check(request, page, slug)
```

Can I remove this argument or deprecate it with warning or there are plans to use it somewhere it the future?

**Comment by markrofail** [comment]
Why is this still open? Wasn't it resolved by #7020

**Comment by marksweb** [comment]
@markrofail Good spot, I'll close this, thanks!
